OrbitLinuxTracing/GpuTracepointEventProcessorTest.cpp

```diff
@@ -0,0 +1,589 @@
+  orbit_grpc_protos::GpuJob actual_gpu_job;
+  EXPECT_CALL(mock_listener_, OnGpuJob).Times(1).WillOnce(::testing::SaveArg<0>(&actual_gpu_job));
+  processor_.PushEvent(
+      MakeFakeAmdgpuSchedRunJobPerfEvent(kTimestampB, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(
+      MakeFakeAmdgpuCsIoctlPerfEvent(kTid, kTimestampA, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(MakeFakeDmaFenceSignaledPerfEvent(kTimestampD, kContext, kSeqno, kTimeline));
+  ExpectGpuJobEq(actual_gpu_job, expected_gpu_job);
+}
+
+TEST_F(GpuTracepointEventProcessorTest, MismatchingContext) {
+  static constexpr pid_t kTid = 42;
+  static constexpr uint32_t kContext = 1;
+  static constexpr uint32_t kSeqno = 10;
+  static const std::string kTimeline = "timeline";
+  static constexpr uint64_t kTimestampA = 100;
+  static constexpr uint64_t kTimestampB = 200;
+  static constexpr uint64_t kTimestampD = 300;
+
+  EXPECT_CALL(mock_listener_, OnGpuJob).Times(0);
+  processor_.PushEvent(
+      MakeFakeAmdgpuCsIoctlPerfEvent(kTid, kTimestampA, kContext + 1, kSeqno, kTimeline));
+  processor_.PushEvent(
+      MakeFakeAmdgpuSchedRunJobPerfEvent(kTimestampB, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(MakeFakeDmaFenceSignaledPerfEvent(kTimestampD, kContext, kSeqno, kTimeline));
+}
+
+TEST_F(GpuTracepointEventProcessorTest, MismatchingSeqno) {
+  static constexpr pid_t kTid = 42;
+  static constexpr uint32_t kContext = 1;
+  static constexpr uint32_t kSeqno = 10;
+  static const std::string kTimeline = "timeline";
+  static constexpr uint64_t kTimestampA = 100;
+  static constexpr uint64_t kTimestampB = 200;
+  static constexpr uint64_t kTimestampD = 300;
+
+  EXPECT_CALL(mock_listener_, OnGpuJob).Times(0);
+  processor_.PushEvent(
+      MakeFakeAmdgpuCsIoctlPerfEvent(kTid, kTimestampA, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(
+      MakeFakeAmdgpuSchedRunJobPerfEvent(kTimestampB, kContext, kSeqno + 1, kTimeline));
+  processor_.PushEvent(MakeFakeDmaFenceSignaledPerfEvent(kTimestampD, kContext, kSeqno, kTimeline));
+}
+
+TEST_F(GpuTracepointEventProcessorTest, MismatchingTimeline) {
+  static constexpr pid_t kTid = 42;
+  static constexpr uint32_t kContext = 1;
+  static constexpr uint32_t kSeqno = 10;
+  static const std::string kTimeline = "timeline";
+  static constexpr uint64_t kTimestampA = 100;
+  static constexpr uint64_t kTimestampB = 200;
+  static constexpr uint64_t kTimestampD = 300;
+
+  EXPECT_CALL(mock_listener_, OnGpuJob).Times(0);
+  processor_.PushEvent(
+      MakeFakeAmdgpuCsIoctlPerfEvent(kTid, kTimestampA, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(
+      MakeFakeAmdgpuSchedRunJobPerfEvent(kTimestampB, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(
+      MakeFakeDmaFenceSignaledPerfEvent(kTimestampD, kContext, kSeqno, kTimeline + "1"));
+}
+
+TEST_F(GpuTracepointEventProcessorTest, TwoNonOverlappingJobsDifferingByContext) {
```

Comment:
Again, it would be nice to get from the name an intuition about what to expect.

Reply:
I added "WithSameDepth" as that's the main factor here. Left out anything like "listener notified" or "jobs created" as they add very little.

Reply:
Sounds good.